haunted-houses/src/main/java/cz/muni/fi/pa165/hauntedhouses/dao/impl/AbilityDaoImpl.java

```diff
@@ -2,76 +2,58 @@
 
 import cz.muni.fi.pa165.hauntedhouses.dao.AbilityDao;
 import cz.muni.fi.pa165.hauntedhouses.model.Ability;
+import org.springframework.stereotype.Repository;
 
 import javax.persistence.*;
+import javax.transaction.Transactional;
 import java.util.List;
 
 /**
  * @author David Hofman
  */
+@Repository
+@Transactional
 public class AbilityDaoImpl implements AbilityDao {
 
-    private static EntityManagerFactory emf = Persistence.createEntityManagerFactory("default");
+    @PersistenceContext
+    private static EntityManager em;
 
     @Override
     public void createAbility(Ability A) {
-        EntityManager em = emf.createEntityManager();
-        em.getTransaction().begin();
         em.persist(A);
-        em.getTransaction().commit();
-        em.close();
     }
 
     @Override
     public Ability getAbilityById(Long id) {
-        EntityManager em = emf.createEntityManager();
-        Ability ability = em.find(Ability.class, id);
-        em.close();
-        return ability;
+        return em.find(Ability.class, id);
     }
 
     @Override
     public Ability getAbilityByName(String name) {
-        EntityManager em = emf.createEntityManager();
-        TypedQuery<Ability> query = em.createQuery("SELECT a FROM Ability a WHERE a.name = :name", Ability.class).setParameter("name",name);
-        Ability ability;
         try {
-            ability = query.getSingleResult();
+            return em.createQuery("SELECT a FROM Ability a WHERE a.name = :name", Ability.class)
+                    .setParameter("name",name).getSingleResult();
         } catch(NoResultException e) {
-            ability = null;
-        } finally {
-            em.close();
+            return null;
         }
-        return ability;
     }
 
     @Override
     public List<Ability> getAllAbilities() {
-        EntityManager em = emf.createEntityManager();
-        List<Ability> list = em.createQuery("SELECT a FROM Ability a", Ability.class).getResultList();
-        em.close();
-        return list;
+        return em.createQuery("SELECT a FROM Ability a", Ability.class).getResultList();
     }
 
     @Override
-    public void updateAbility(Ability A) {
-        EntityManager em = emf.createEntityManager();
-        em.getTransaction().begin();
-        em.merge(A);
-        em.getTransaction().commit();
-        em.close();
+    public Ability updateAbility(Ability A) {
+        return em.merge(A);
```

Comment:
Chceme teda sem pridať if condition, aby sme najskôr otestovali, či tá entita tam je?

Reply:
Pokud nechceme, aby se případně vytvořila nová, tak asi ano.

Reply:
Já bych to tam přidal, jinak bude metoda update přidávat nové entity a to asi nechceme.

Reply:
Ok, tak tam pridaj if a vracaj null, ak ta entita nebude existovat